packages/core/src/Engine.ts

```diff
@@ -141,15 +145,28 @@ export class Engine extends EventDispatcher {
   private _waitingGC: boolean = false;
 
   private _animate = () => {
-    if (this._vSyncCount) {
-      this._requestId = requestAnimationFrame(this._animate);
-      if (this._vSyncCounter++ % this._vSyncCount === 0) {
+    const { xrManager } = this;
+    if (xrManager) {
+      this._requestId = xrManager.sessionManager.requestAnimationFrame(this._animate);
+      if (this._vSyncCount) {
+        if (this._vSyncCounter++ % this._vSyncCount === 0) {
+          this.update();
+          this._vSyncCounter = 1;
+        }
+      } else {
         this.update();
-        this._vSyncCounter = 1;
       }
     } else {
-      this._timeoutId = window.setTimeout(this._animate, this._targetFrameInterval);
-      this.update();
+      this._requestId = window.requestAnimationFrame(this._animate);
+      if (this._vSyncCount) {
+        if (this._vSyncCounter++ % this._vSyncCount === 0) {
+          this.update();
+          this._vSyncCounter = 1;
+        }
+      } else {
+        this._timeoutId = window.setTimeout(this._animate, this._targetFrameInterval);
+        this.update();
+      }
     }
   };
```

Comment:
This is OK? Or you have other considerations？
```ts
private _animate = () => {
    if (this._vSyncCount) {
      const { xrManager } = this;
      this._requestId = xrManager
        ? xrManager.sessionManager.requestAnimationFrame(this._animate)
        : window.requestAnimationFrame(this._animate);
      if (this._vSyncCounter++ % this._vSyncCount === 0) {
        this.update();
        this._vSyncCounter = 1;
      }
    } else {
      this._timeoutId = window.setTimeout(this._animate, this._targetFrameInterval);
      this.update();
    }
  };
```

Reply:
We should consider XR first and then settimeout.